source/interprocedural_analyses/taint/modelParser.ml

```diff
@@ -2995,33 +3053,36 @@ let rec parse_statement
                in
 
                List.concat_map body ~f:signature)
-        |> Option.value ~default:[]
-        |> return
+        |> function
+        | Some result -> List.map ~f:(fun res -> Ok res) result
+        | None -> []
```

Comment:
```suggestion
        |> Option.bind ~f:(List.map ~f:return)
        |> Option.value ~default:[]
```

Reply:
@arthaud this gives type error for the `List.map` function used by Option.bind.
We could reverse back to the match clauses or use the following new code that I imagine will work:
```Ocaml
         |> Option.value ~default:[]
         |> List.map ~f:return
```